File: S/1945-sum-of-digits-of-string-after-convert/main.go
package main

import "fmt"

func getLucky(s string, k int) int {
	buf := []int{}
	for _, b := range s {
		buf = append(buf, int(byte(b)-'a')+1)
	}
	sum := 0
	for i := 0; i < len(buf); i++ {
		c := buf[i]
		for c > 0 {
			sum += c % 10
			c /= 10
		}
	}

	for i := 1; i < k; i++ {
		temp := sum
		t := 0
		for temp > 0 {
			t += temp % 10
			temp /= 10
		}
		sum = t
	}
	return sum
}

func main() {
	fmt.Println()
}

File: S/1945-sum-of-digits-of-string-after-convert/main.rs
struct Solution {}

impl Solution {
  pub fn get_lucky(s: String, k: i32) -> i32 {
    let mut sum: i32 = 0;
    for b in s.as_bytes() {
      let mut offset = (b - 'a' as u8) as i32 + 1;
      while offset > 0 {
        sum += offset % 10;
        offset /= 10;
      }
    }

    for i in 1..k {
      let mut t = sum;
      let mut sum2: i32 = 0;
      while t > 0 {
        sum2 += t % 10;
        t /= 10;
      }
      sum = sum2;
    }
    sum
  }
}
